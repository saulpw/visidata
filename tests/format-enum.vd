sheet	col	row	longname	input	keystrokes	comment
			open-file	sample_data/sample.tsv	o	
sample	OrderDate		type-date		@	set type of current column to date
sample	Units		type-int		#	set type of current column to int
sample	Unit_Cost		type-float		%	set type of current column to float
sample			columns-sheet		Shift+C	open Columns Sheet: edit column properties for current sheet
sample_columns	fmtstr	ã‚­OrderDate	edit-cell	%Y	e	edit contents of current cell
sample	Item	0	setcol-format-enum	Pencil=1 Binder=2 Pen=3 Desk=4		add secondary type translator to current column from input enum (space-separated)
