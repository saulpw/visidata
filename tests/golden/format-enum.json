[
{"OrderDate": "2016", "Region": "East", "Rep": "Jones", "Item": "Pencil", "Units": 95, "Unit_Cost": 1.99, "Total": "189.05"},
{"OrderDate": "2016", "Region": "Central", "Rep": "Kivell", "Item": "Binder", "Units": 50, "Unit_Cost": 19.99, "Total": "999.50"},
{"OrderDate": "2016", "Region": "Central", "Rep": "Jardine", "Item": "Pencil", "Units": 36, "Unit_Cost": 4.99, "Total": "179.64"},
{"OrderDate": "2016", "Region": "Central", "Rep": "Gill", "Item": "Pen", "Units": 27, "Unit_Cost": 19.99, "Total": "539.73"},
{"OrderDate": "2016", "Region": "West", "Rep": "Sorvino", "Item": "Pencil", "Units": 56, "Unit_Cost": 2.99, "Total": "167.44"},
{"OrderDate": "2016", "Region": "East", "Rep": "Jones", "Item": "Binder", "Units": 60, "Unit_Cost": 4.99, "Total": "299.40"},
{"OrderDate": "2016", "Region": "Central", "Rep": "Andrews", "Item": "Pencil", "Units": 75, "Unit_Cost": 1.99, "Total": "149.25"},
{"OrderDate": "2016", "Region": "Central", "Rep": "Jardine", "Item": "Pencil", "Units": 90, "Unit_Cost": 4.99, "Total": "449.10"},
{"OrderDate": "2016", "Region": "West", "Rep": "Thompson", "Item": "Pencil", "Units": 32, "Unit_Cost": 1.99, "Total": "63.68"},
{"OrderDate": "2016", "Region": "East", "Rep": "Jones", "Item": "Binder", "Units": 60, "Unit_Cost": 8.99, "Total": "539.40"},
{"OrderDate": "2016", "Region": "Central", "Rep": "Morgan", "Item": "Pencil", "Units": 90, "Unit_Cost": 4.99, "Total": "449.10"},
{"OrderDate": "2016", "Region": "East", "Rep": "Howard", "Item": "Binder", "Units": 29, "Unit_Cost": 1.99, "Total": "57.71"},
{"OrderDate": "2016", "Region": "East", "Rep": "Parent", "Item": "Binder", "Units": 81, "Unit_Cost": 19.99, "Total": "1619.19"},
{"OrderDate": "2016", "Region": "East", "Rep": "Jones", "Item": "Pencil", "Units": 35, "Unit_Cost": 4.99, "Total": "174.65"},
{"OrderDate": "2016", "Region": "Central", "Rep": "Smith", "Item": "Desk", "Units": 2, "Unit_Cost": 125.0, "Total": "250.00"},
{"OrderDate": "2016", "Region": "East", "Rep": "Jones", "Item": "Pen Set", "Units": 16, "Unit_Cost": 15.99, "Total": "255.84"},
{"OrderDate": "2016", "Region": "Central", "Rep": "Morgan", "Item": "Binder", "Units": 28, "Unit_Cost": 8.99, "Total": "251.72"},
{"OrderDate": "2016", "Region": "East", "Rep": "Jones", "Item": "Pen", "Units": 64, "Unit_Cost": 8.99, "Total": "575.36"},
{"OrderDate": "2016", "Region": "East", "Rep": "Parent", "Item": "Pen", "Units": 15, "Unit_Cost": 19.99, "Total": "299.85"},
{"OrderDate": "2016", "Region": "Central", "Rep": "Kivell", "Item": "Pen Set", "Units": 96, "Unit_Cost": 4.99, "Total": "479.04"},
{"OrderDate": "2016", "Region": "Central", "Rep": "Smith", "Item": "Pencil", "Units": 67, "Unit_Cost": 1.29, "Total": "86.43"},
{"OrderDate": "2016", "Region": "East", "Rep": "Parent", "Item": "Pen Set", "Units": 74, "Unit_Cost": 15.99, "Total": "1183.26"},
{"OrderDate": "2017", "Region": "Central", "Rep": "Gill", "Item": "Binder", "Units": 46, "Unit_Cost": 8.99, "Total": "413.54"},
{"OrderDate": "2017", "Region": "Central", "Rep": "Smith", "Item": "Binder", "Units": 87, "Unit_Cost": 15.0, "Total": "1305.00"},
{"OrderDate": "2017", "Region": "East", "Rep": "Jones", "Item": "Binder", "Units": 4, "Unit_Cost": 4.99, "Total": "19.96"},
{"OrderDate": "2017", "Region": "West", "Rep": "Sorvino", "Item": "Binder", "Units": 7, "Unit_Cost": 19.99, "Total": "139.93"},
{"OrderDate": "2017", "Region": "Central", "Rep": "Jardine", "Item": "Pen Set", "Units": 50, "Unit_Cost": 4.99, "Total": "249.50"},
{"OrderDate": "2017", "Region": "Central", "Rep": "Andrews", "Item": "Pencil", "Units": 66, "Unit_Cost": 1.99, "Total": "131.34"},
{"OrderDate": "2017", "Region": "East", "Rep": "Howard", "Item": "Pen", "Units": 96, "Unit_Cost": 4.99, "Total": "479.04"},
{"OrderDate": "2017", "Region": "Central", "Rep": "Gill", "Item": "Pencil", "Units": 53, "Unit_Cost": 1.29, "Total": "68.37"},
{"OrderDate": "2017", "Region": "Central", "Rep": "Gill", "Item": "Binder", "Units": 80, "Unit_Cost": 8.99, "Total": "719.20"},
{"OrderDate": "2017", "Region": "Central", "Rep": "Kivell", "Item": "Desk", "Units": 5, "Unit_Cost": 125.0, "Total": "625.00"},
{"OrderDate": "2017", "Region": "East", "Rep": "Jones", "Item": "Pen Set", "Units": 62, "Unit_Cost": 4.99, "Total": "309.38"},
{"OrderDate": "2017", "Region": "Central", "Rep": "Morgan", "Item": "Pen Set", "Units": 55, "Unit_Cost": 12.49, "Total": "686.95"},
{"OrderDate": "2017", "Region": "Central", "Rep": "Kivell", "Item": "Pen Set", "Units": 42, "Unit_Cost": 23.95, "Total": "1005.90"},
{"OrderDate": "2017", "Region": "West", "Rep": "Sorvino", "Item": "Desk", "Units": 3, "Unit_Cost": 275.0, "Total": "825.00"},
{"OrderDate": "2017", "Region": "Central", "Rep": "Gill", "Item": "Pencil", "Units": 7, "Unit_Cost": 1.29, "Total": "9.03"},
{"OrderDate": "2017", "Region": "West", "Rep": "Sorvino", "Item": "Pen", "Units": 76, "Unit_Cost": 1.99, "Total": "151.24"},
{"OrderDate": "2017", "Region": "West", "Rep": "Thompson", "Item": "Binder", "Units": 57, "Unit_Cost": 19.99, "Total": "1139.43"},
{"OrderDate": "2017", "Region": "Central", "Rep": "Andrews", "Item": "Pencil", "Units": 14, "Unit_Cost": 1.29, "Total": "18.06"},
{"OrderDate": "2017", "Region": "Central", "Rep": "Jardine", "Item": "Binder", "Units": 11, "Unit_Cost": 4.99, "Total": "54.89"},
{"OrderDate": "2017", "Region": "Central", "Rep": "Jardine", "Item": "Binder", "Units": 94, "Unit_Cost": 19.99, "Total": "1879.06"},
{"OrderDate": "2017", "Region": "Central", "Rep": "Andrews", "Item": "Binder", "Units": 28, "Unit_Cost": 4.99, "Total": "139.72"}
]
